### PR TITLE
Wishlist api

### PR DESCRIPTION
**Pull request by VedanthB**
- add to wish list
- get wishlist  items 
- add to cart from wishlist 
- remove from wishlist 
- add to wishlist from cart

### CODE REVIEW

File: src/components/auth-pages/LoginCard.js [hunk 1]
```diff
@@ -1,84 +1,130 @@
...
+        setUserData({ ...userData, [name]: value })
+    }
 
-        <Link
-          to="/"
-          className="btn btn-link-cyan text-white mb-5 pl-0"
-          style={{ padding: 0 }}
+    return (
+        <form
+            style={{ margin: 'auto', backgroundColor: '#0000227c' }}
```
Comment (Ananyamadhu08) [inline src/components/auth-pages/LoginCard.js:25]: you can define a class for margin auto and variable for colour.

File: src/context/providers/WishlistProvider.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+    const { encodedToken } = authState
+
+    useEffect(() => {
+        if (encodedToken) {
+            getWishListItems(encodedToken, wishlistDispatch)
+        }
+    }, [encodedToken])
+
+    console.log(wishlistState)
```
Comment (Ananyamadhu08) [inline src/context/providers/WishlistProvider.js:32]: can delete the console logs

File: src/context/reducers/filterReducer.js [hunk 3]
```diff
@@ -0,0 +1,51 @@
+import { filtersConstants } from '../constants/filtersConstants'
+
+// sortBy: null,
```
Comment (Ananyamadhu08) [inline src/context/reducers/filterReducer.js:3]: can deleted commented code

**Review by Ananyamadhu08 — APPROVED** [review]
You can delete the commented code and also define classes for repetitive things such as margin auto. Other than that your code is great, good job you can go ahead and merge.

File: src/utils/addWishlistItem.js [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+            status,
+        } = await postWishlistItemService(token, item)
+
+        if (status >= 200 && status < 300) {
+            wishlistDispatch({
+                type: wishlistActions.ADD_WISHLIST_ITEM,
+                payload: wishlist,
+            })
+
+            console.log(wishlist, status)
```
Comment (prathmeshjagtap) [inline src/utils/addWishlistItem.js:19]: remove console.log

File: src/utils/filterProducts.js [hunk 5]
```diff
@@ -0,0 +1,54 @@
+export const filterByCategory = (products, categoriesFilter) => {
+    let filterProducts = []
+
+    const noFilterSelected = Object.values(categoriesFilter).every(
+        (category) => !category
+    )
+
+    console.log(noFilterSelected)
+
```
Comment (prathmeshjagtap) [inline src/utils/filterProducts.js:9]: remove console.log

File: src/utils/filterProducts.js [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+export const filterByBrands = (products, brandsFilter) => {
+    let filterProducts = []
+
+    const noFilterSelected = Object.values(brandsFilter).every(
+        (brand) => !brand
+    )
+
+    console.log(noFilterSelected)
+
```
Comment (prathmeshjagtap) [inline src/utils/filterProducts.js:34]: remove console.log

File: src/utils/filterProducts.js [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+        }
+
+        filterProducts = [...filterProducts, ...filteredProductsByBrand]
+    }
+
+    return filterProducts
+}
+
+export const filterByRatings = (products, ratingsLessThan) => {
+    console.log(ratingsLessThan)
```
Comment (prathmeshjagtap) [inline src/utils/filterProducts.js:52]: remove console.log

File: src/utils/getProducts.js [hunk 8]
```diff
@@ -0,0 +1,34 @@
+import axios from 'axios'
+
+import { productActions } from '../context/constants/productConstants'
+import { useProducts } from '../context/providers/ProductProvider'
+import { getProductsService } from '../services/productServices'
+
+export const getProducts = async (productsDispatch) => {
+    try {
+        console.log(productsDispatch)
+
```
Comment (prathmeshjagtap) [inline src/utils/getProducts.js:10]: remove console.log

**Review by prathmeshjagtap — APPROVED** [review]

**Pull request merged**